apps/src/calc/inputIterator.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Given a set of values (i.e. [1,2,3], and a number of parameters, generates
+ * all possible combinations of values.
+ */
+var InputIterator = function (values, numParams) {
```

Comment:
Interesting. My original solution looked something like the top answer on the SO article, and I decided it was pretty difficult to read/parse. The advantage to this solution I thought was (a) easier to read/understand and (b) lazy generation meant we didn't take the hit for inputs we didn't use (i.e. we'll stop as soon as one fails). I'm not sure that (b) is really all that compelling since the non-failure case is more common, so if it's expensive to generate all inputs, we're already in trouble.

Do you think it makes sense to move from the existing generator to some sort of cartesian product? I'm inclined to stick with what I have, given that it works.